mirascope/core/base/prompt.py

```diff
@@ -307,18 +309,24 @@ async def run():
 _BasePromptT = TypeVar("_BasePromptT", bound=BasePrompt)
 _MessageParamT = TypeVar("_MessageParamT", bound=Any)
 _BaseDynamicConfigT = TypeVar("_BaseDynamicConfigT", bound=BaseDynamicConfig)
+_MessageFuncReturnTypes: TypeAlias = (
+    str | BaseMessageParam | list[BaseMessageParam] | list[Any]
+)
+_MessageFuncReturnT = TypeVar(
+    "_MessageFuncReturnT", bound=_MessageFuncReturnTypes, contravariant=True
+)
 
 
 class PromptDecorator(Protocol):
     @overload
     def __call__(
         self, prompt: Callable[_P, BaseDynamicConfig]
-    ) -> Callable[_P, list[BaseMessageParam]]: ...
+    ) -> Callable[_P, Messages]: ...
 
```

Comment:
Checked if `list[BaseMessageParam]` and `Messages ` can be used to mean the same thing.